libraries/ui-shared/ScannerOverlay/index.jsx

```diff
@@ -1,61 +1,32 @@
-import React, { Fragment, PureComponent } from 'react';
+import React, { Fragment } from 'react';
 import PropTypes from 'prop-types';
-import AppScanner from '@shopgate/pwa-core/classes/Scanner';
 import CameraOverlay from './components/CameraOverlay';
 import ScannerBar from './components/ScannerBar';
+import FlashlightButton from './components/ScannerBar/components/FlashlightButton';
+
+export { CameraOverlay, ScannerBar, FlashlightButton };
 
 /**
  * The scanner overlay component.
+ * @param {Function} cameraOverlay camera overlay with corners
+ * @param {Function} scannerBar scanner bar
+ * @returns {JSX}
  */
-class ScannerOverlay extends PureComponent {
-  static propTypes = {
-    flashlight: PropTypes.bool,
-  }
-
-  static defaultProps = {
-    flashlight: false,
-  }
-
-  /**
-   * Initializes the component.
-   * @param {Object} props The components props.
-   */
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      flashlight: props.flashlight,
-    };
-  }
-
-  /**
-   * @param {Object} nextProps New props to apply.
-   */
-  componentWillReceiveProps(nextProps) {
-    if (this.state.flashlight !== nextProps.flashlight) {
-      this.setState({ flashlight: nextProps.flashlight });
-    }
-  }
-
-  handleToggleFlashlight = () => {
-    this.setState({ flashlight: AppScanner.toggleFlashlight() });
-  }
-
-  /**
-   * Render the camera overlay and the bottom bar with its contents.
-   * @returns {JSX}
-   */
-  render() {
-    return (
-      <Fragment>
-        <CameraOverlay />
-        <ScannerBar
-          flashlightState={this.state.flashlight}
-          onToggleFlashlight={this.handleToggleFlashlight}
-        />
-      </Fragment>
-    );
-  }
-}
+const ScannerOverlay = ({ cameraOverlay, scannerBar }) => (
```

Comment:
What is that component? This is unnecessary in my opinion. There is no use case for it. It just outputs the two parameters that it receives with fallback values. These things can also be output at the place of usage.